configure.ac

```diff
@@ -219,24 +219,38 @@ AC_CHECK_LIB([gsl],[gsl_sf_bessel_Jn], [],
 # Check for HDF5 library
 AC_ARG_WITH(hdf5, [AC_HELP_STRING([--without-hdf5],[do not allow HDF5 output])], ok=$withval, ok=yes)
 if test "$ok" = "yes"; then
-	AC_CHECK_LIB(z,deflate, [],[AC_MSG_WARN([zlib is required for HDF5!])])
-
-	save_CC=$CC
-	if test "x$with_mpi" = "xyes"; then
-	   CC=$MPICXX
-	fi
-	save_LIBS_0="$LIBS" # save, to check later if we found any library
-	AC_CHECK_LIB(hdf5, H5Pcreate, [
-		AC_CHECK_HEADERS(hdf5.h, [LIBS="-lhdf5 $LIBS"
-                     AC_DEFINE(HAVE_HDF5,1,[Define if we have & link HDF5])])])
-	if test x"$save_LIBS_0" = x"$LIBS"; then
-		AC_MSG_WARN([Couldn't find the HDF5 library!!  Switching to --without-hdf5.])
-	fi
-	CC=$save_CC
-
-        if test "x$with_mpi" = "xyes"; then
-		AC_CHECK_FUNCS(H5Pset_mpi H5Pset_fapl_mpio)
-	fi
+    AC_CHECK_LIB(z,deflate, [],[AC_MSG_WARN([zlib is required for HDF5!])])
+
+    save_CC="$CC"
+    save_CPPFLAGS="$CPPFLAGS"
+    save_LIBS="$LIBS"
+
+    # Attempt to locate HDF5 via pkg-config and fall back to a default.
+    PKG_CHECK_MODULES(HDF5, hdf5, [
+        CPPFLAGS="$CPPFLAGS $HDF5_CFLAGS"
+        LIBS="$HDF5_LIBS $LIBS"
+    ], [
+        LIBS="-lhdf5 $LIBS"
+    ])
+
+    if test "x$with_mpi" = "xyes"; then
+        CC=$MPICPP
```

Comment:
This rename doesn't seem right

Reply:
(This is a user-visible name, people use it in their scripts to specify the MPI C++ compiler, and it's also used by the AX_MPI macro that we use.)

Reply:
If github uses `$MPICPP`, then just pass `MPICXX=$MPICPP` to the `configure` script.

Reply:
Yes, this should be CC=$MPICXX, just like in the previous implementation.
I will push a commit to fix this.